### PR TITLE
Runtime stride

### PR DESCRIPTION
**Pull request by dlangbe**
Added the ability to change the default stride layout at runtime with environment variable `HIPTENSOR_DEFAULT_STRIDES_COL_MAJOR`. 

Renamed `HIPTENSOR_DATA_LAYOUT_COL_MAJOR -> HIPTENSOR_DEFAULT_STRIDES_COL_MAJOR`

### CODE REVIEW

**Comment by amd-jnovotny** [comment]
@dlangbe Is there anything to add to the changelog for this?

**Comment by dlangbe** [comment]
> @dlangbe Is there anything to add to the changelog for this?

Yes, the changelog should reflect the naming change of `HIPTENSOR_DATA_LAYOUT_COL_MAJOR -> HIPTENSOR_DEFAULT_STRIDES_COL_MAJOR`

**Comment by amd-jnovotny** [comment]
@dlangbe Thanks for the clarification. After you update the change log, I can review it for formatting, etc.

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ Full documentation for hipTensor is available at [rocm.docs.amd.com/projects/hip
 * ASAN library builds now use -mcmodel=large to accommodate larger lib size
 * Updated permute backend to accommodate changes to element-wise ops implementation
 * Updated validation acceptance criteria to match CK backend tests
+* Renamed CMake option HIPTENSOR_DATA_LAYOUT_COL_MAJOR -> HIPTENSOR_DEFAULT_STRIDES_COL_MAJOR
```
Comment (amd-jnovotny) [inline CHANGELOG.md]:
```suggestion
* Renamed the CMake option `HIPTENSOR_DATA_LAYOUT_COL_MAJOR` to `HIPTENSOR_DEFAULT_STRIDES_COL_MAJOR`
```

**Review by amd-jnovotny — COMMENTED** [review]
Just some formatting suggestions and a slight rewording for the changelog. Looks good otherwise. Thank you!

File: CMakeLists.txt [hunk 2]
```diff
@@ -66,7 +66,7 @@ if( CMAKE_PROJECT_NAME STREQUAL "hiptensor" )
   option( HIPTENSOR_BUILD_TESTS "Build hiptensor tests" ON )
   option( HIPTENSOR_BUILD_SAMPLES "Build hiptensor samples" ON )
   option( HIPTENSOR_BUILD_COMPRESSED_DBG "Enable compressed debug symbols" ON)
-  option( HIPTENSOR_DATA_LAYOUT_COL_MAJOR "Set hiptensor data layout to column major" ON )
+  option( HIPTENSOR_DEFAULT_STRIDES_COL_MAJOR "Set hiptensor data layout to column major" ON )
```
Comment (cgmillette) [inline CMakeLists.txt]: Can we update the description here to :  "Set hipTensor default strides to column major"

File: library/src/hiptensor_options.cpp [hunk 3]
```diff
@@ -41,7 +41,26 @@ namespace hiptensor
         , mColdRuns(0)
         , mInputFilename("")
         , mOutputFilename("")
+        , mColMajorStrides(true)
```
Comment (cgmillette) [inline library/src/hiptensor_options.cpp]: Should maybe initialize this to the existing value of HIPTENSOR_DEFAULT_STRIDES_COL_MAJOR symbol that was set during cmake. Then we can remove the 'else' clause below

**Review by cgmillette — COMMENTED** [review]
Couple minute changes, but otherwise looks great!

**Review by cgmillette — APPROVED** [review]
LGTM

**Pull request merged**